### PR TITLE
Add a wrapper for accessing the URL launcher service.

### PR DESCRIPTION
**Pull request by chinmaygarde**
cc @HansMuller 

https://gist.github.com/chinmaygarde/99b3c6bc700a973a51639526d835319f if anyone needs the snippet.

### CODE REVIEW

File: packages/flutter/lib/src/services/uri_launcher.dart [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  ///   before it attempts to launch the same.
+  ///
+  /// Return Value:
+  ///
+  ///   boolean indicating if the intent to handle the URI was successfully
+  ///   conveyed to the to underlying platform and the platform could
+  ///   successfully handle the same. The platform is responsible for URI
+  ///   parsing.
+  static Future<bool> launch(String uriString) {
+    return _connectedURILauncherService.launch(uriString);
```
Comment (abarth) [inline packages/flutter/lib/src/services/uri_launcher.dart]: There's a bit more ceremony required here.  See the other wrappers for the pattern.

**Comment by chinmaygarde** [comment]
I see, I didn't pull in the changes made recently and was basing this off of an older pattern https://github.com/flutter/flutter/commit/a8f6f44a5539ba4c11a1e76bc932a5031f60a260#diff-7d620ba5e4aa11648c48a973a928da2fR33

File: packages/flutter/lib/src/services/uri_launcher.dart [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  /// * [uriString]: The URI string to be parsed by the underlying platform and
+  ///   before it attempts to launch the same.
+  ///
+  /// Return Value:
+  ///
+  ///   boolean indicating if the intent to handle the URI was successfully
+  ///   conveyed to the to underlying platform and the platform could
+  ///   successfully handle the same. The platform is responsible for URI
+  ///   parsing.
+  static Future<bool> launch(String uriString) {
```
Comment (HansMuller) [inline packages/flutter/lib/src/services/uri_launcher.dart]: I think we're going to want to use the term URL instead of URI. Ask @Hixie about that.

**Comment by chinmaygarde** [comment]
Android will parse the string as conforming to RFC 2396 while iOS expects the string to conform to RFC 1738. I believe 2396 updates 1738 and RFC 3986 updates or obsoletes both. So depending on the platform, there will be subtle changes in the conformance requirements of the string. That is why the mojom specifies that the platform will parse this instead of us. I associate URL with 1738 and URI with the others.

**Comment by Hixie** [comment]
The URL standard obsoletes RFC 3986, RFC 2396, and RFC 1738.

**Comment by chinmaygarde** [comment]
Done. Another look?

File: packages/flutter/lib/src/services/url_launcher.dart [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  return shell.connectToApplicationService(
+      mojo.UrlLauncher.serviceName,
+      mojo.UrlLauncher.connectToService);
+}
+
+final mojo.UrlLauncherProxy _connectedURLLauncherService = _initURLLauncherProxy();
+
+/// Allows applications to delegate responsbility of handling certain URLs to
+/// the underlying platform.
+class URLLauncher {
```
Comment (abarth) [inline packages/flutter/lib/src/services/url_launcher.dart]: a/URLLauncher/UrlLauncher/

File: packages/flutter/lib/src/services/url_launcher.dart [hunk 4]
```diff
@@ -0,0 +1,44 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:sky_services/flutter/platform/url_launcher.mojom.dart' as mojo;
+import 'shell.dart';
+
+mojo.UrlLauncherProxy _initURLLauncherProxy() {
```
Comment (abarth) [inline packages/flutter/lib/src/services/url_launcher.dart]: s/_initURLLauncherProxy/_initUrlLauncherProxy/

(These are from the Dart style guide.)

File: packages/flutter/lib/src/services/url_launcher.dart [hunk 5]
```diff
@@ -0,0 +1,44 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:sky_services/flutter/platform/url_launcher.mojom.dart' as mojo;
```
Comment (abarth) [inline packages/flutter/lib/src/services/url_launcher.dart]: We usually import mojom files into a `mojom` namespace.

**Comment by abarth** [comment]
LGTM

**Comment by chinmaygarde** [comment]
Done. Rebased.

**Pull request merged**